web-filter-base/src/com/untangle/app/web_filter/WebFilterDecisionEngine.java

```diff
@@ -139,7 +140,16 @@ public HttpRedirect checkRequest(AppTCPSession sess, InetAddress clientIp, int p
                      */
                     if (matcherO == null || !(matcherO instanceof GlobMatcher)) {
                         try{
-                            matcher = GlobMatcher.getMatcher("*\\b" + rule.getString() + "\\b*");
+                            //Code to handle special characters in search terms
+                            String match = null;
+                            Pattern specialCharacter = Pattern.compile (".*[!@#$%&*()_+=|<>?{}\\[\\]~-].*");
```

Comment:
For performance, make specialCharacter a static patten.

Reply:
Done as suggested